master/src/main/scala/org/apache/celeborn/service/deploy/master/network/CelebornRackResolver.scala

```diff
@@ -17,33 +17,80 @@
 
 package org.apache.celeborn.service.deploy.master.network
 
+import java.io.File
+import java.util.concurrent.{ScheduledExecutorService, TimeUnit}
+
 import scala.collection.JavaConverters._
 import scala.collection.mutable.ArrayBuffer
 
 import com.google.common.base.Strings
 import org.apache.hadoop.conf.Configuration
 import org.apache.hadoop.fs.CommonConfigurationKeysPublic
-import org.apache.hadoop.net.{CachedDNSToSwitchMapping, DNSToSwitchMapping, NetworkTopology, Node, NodeBase, ScriptBasedMapping}
+import org.apache.hadoop.net.{CachedDNSToSwitchMapping, DNSToSwitchMapping, NetworkTopology, Node, NodeBase, ScriptBasedMapping, TableMapping}
 import org.apache.hadoop.util.ReflectionUtils
 
 import org.apache.celeborn.common.CelebornConf
 import org.apache.celeborn.common.internal.Logging
-import org.apache.celeborn.common.util.CelebornHadoopUtils
+import org.apache.celeborn.common.util.{CelebornHadoopUtils, ThreadUtils}
 
 class CelebornRackResolver(celebornConf: CelebornConf) extends Logging {
 
+  private var rackResolveRefreshThreadPool: ScheduledExecutorService = _
+
+  private var rackResolveLastModifiedTime = 0L
+
   private val dnsToSwitchMapping: DNSToSwitchMapping = {
     val conf: Configuration = CelebornHadoopUtils.newConfiguration(celebornConf)
     val dnsToSwitchMappingClass =
       conf.getClass(
         CommonConfigurationKeysPublic.NET_TOPOLOGY_NODE_SWITCH_MAPPING_IMPL_KEY,
         classOf[ScriptBasedMapping],
         classOf[DNSToSwitchMapping])
-    ReflectionUtils.newInstance(dnsToSwitchMappingClass, conf)
+    val mapping = ReflectionUtils.newInstance(dnsToSwitchMappingClass, conf)
       .asInstanceOf[DNSToSwitchMapping] match {
       case c: CachedDNSToSwitchMapping => c
       case o => new CachedDNSToSwitchMapping(o)
     }
+    val refreshInterval = celebornConf.rackResolveRefreshInterval
+    rackResolveRefreshThreadPool =
+      ThreadUtils.newDaemonSingleThreadScheduledExecutor("rack-check-thread")
+    val fileName = mapping match {
```

Comment:
`mapping` will always be `CachedDNSToSwatchMapping`.
And so the `match` below will always result in `fileName` being `null`.

Suggest to split this into two - and use `switchMapping` (as an ex) in the match.

```suggestion
    val switchMapping = ReflectionUtils.newInstance(dnsToSwitchMappingClass, conf)
      .asInstanceOf[DNSToSwitchMapping] 
    val mapping = switchMapping  match {
      case c: CachedDNSToSwitchMapping => c
      case o => new CachedDNSToSwitchMapping(o)
    }
    val refreshInterval = celebornConf.rackResolveRefreshInterval
    rackResolveRefreshThreadPool =
      ThreadUtils.newDaemonSingleThreadScheduledExecutor("rack-check-thread")
    val fileName = switchMapping match {
```